tools/toolchains/gcc.py

```diff
@@ -188,7 +209,13 @@ def assemble(self, source, object, includes):
     @hook_tool
     def compile(self, cc, source, object, includes):
         # Build compile command
-        cmd = cc + self.get_compile_options(self.get_symbols(), includes)
+        cmd = cc
+
+        # Add coverage flag on files on which coverage is enabled.
+        if self.check_if_coverage_enabled(source):
+            cmd += self.COVERAGE_FLAGS
+
+        cmd += self.get_compile_options(self.get_symbols(), includes)
```

Comment:
@mazimkhan Specifically this kind of logic. Before this block looked like this:

```
# Add coverage flag on files on which coverage is enabled.
if self.check_if_coverage_enabled(source):
    cmd = cc + self.COVERAGE_FLAGS + self.get_compile_options(self.get_symbols(), includes)
else:
    cmd = cc + self.get_compile_options(self.get_symbols(), includes)
```

And then I suggested changing it to what it currently is now, might be worth trying the previous method (there's 3 blocks of code that would do something similar, you'd probably want to revert all three.)